feat: add popular collections search

shahin-hq opened the pull request:
<!--
Thanks for your interest in the project. Bugs filed and PRs submitted are appreciated!

Please make sure you're familiar with and follow the instructions in the [contributing guidelines](https://ark.dev/docs/program-incentives/guidelines/contributing).

Please fill out the information below to expedite the review and (hopefully) merge of your pull request!
-->

## Summary
Depends on #552 

Closes: https://app.clickup.com/t/86dqpda6z

- adds search
- fixes minor issue with pagination
- fixes nominate collection name search bug 

<!-- What changes are being made? -->

<!-- Why are these changes necessary? -->

<!-- How were these changes implemented? -->

## Checklist

<!-- Have you done all of these things (where applicable)?  -->

-   [x] I checked my UI changes against the design and there are no notable differences
-   [x] I checked my UI changes for any responsiveness issues
-   [x] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] I added a storybook entry for the component that was added _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [ ] Ready to be merged

<!-- Feel free to add additional comments. -->

crnkovic reviewed (changes requested):
Since you tackled pagination in this PR, can you also take care of changing the per-page items in this PR? Changing them still does nothing. On that note, I have created a task to fix the bug where next/prev/first/last buttons in the pagination are not applying the query strings.

---

Search looks good, just expand the width a bit so the text fits. Also, can you make it preserve the state so it doesn't lose input focus after searching?

![Screenshot 2023-12-13 at 09 59 25](https://github.com/ArdentHQ/dashbrd/assets/6536260/d6b1a39a-7cf4-4a2b-8517-8c33eab300a8)

Note for @samharperpittam: currently, there are no filters on mobile, so I have created a card to add them to this page.

----

Looks like the table headers are missing when there are no results:

![Screenshot 2023-12-13 at 10 03 04](https://github.com/ArdentHQ/dashbrd/assets/6536260/0ab03ecb-61ca-4440-8a1f-10f805324bbd)

Design:
![Screenshot 2023-12-13 at 10 02 42](https://github.com/ArdentHQ/dashbrd/assets/6536260/e097f504-1a88-48c0-986e-6c2017c4cc9f)

shahin-hq commented:
@crnkovic 
>  I have created a task to fix the bug where next/prev/first/last buttons in the pagination are not applying the query strings.

one line, fixed it

> Search looks good, just expand the width a bit so the text fits. Also, can you make it preserve the state so it doesn't lose input focus after searching?

fixed

> Looks like the table headers are missing when there are no results:

fixed

> can you also take care of changing the per-page items in this PR?

I'll already handled it in the next PR, didn't want to overload this one too much, I can move my changes here if you really want to
